src/cosmicqc/frame.py

```diff
@@ -591,24 +604,87 @@ def find_image_columns(self: CytoDataFrame_type) -> bool:
         ]
 
     @staticmethod
+    def draw_outline_on_image(actual_image_path: str, mask_image_path: str) -> Image:
+        """
+        Draws outlines on a TIFF image based on a mask image and returns
+        the combined result.
+
+        This method takes the path to a TIFF image and a mask image, creates
+        an outline from the mask, and overlays it on the TIFF image. The resulting
+        image, which combines the TIFF image with the mask outline, is returned.
+
+        Args:
+            actual_image_path (str): Path to the TIFF image file.
+            mask_image_path (str): Path to the mask image file.
+
+        Returns:
+            PIL.Image.Image: A PIL Image object that is the result of
+            combining the TIFF image with the mask outline.
+
+        Raises:
+            FileNotFoundError: If the specified image or mask file does not exist.
+            ValueError: If the images are not in compatible formats or sizes.
+        """
+        # Load the TIFF image and convert it to grayscale array
+        tiff_image_array = skimage.io.imread(actual_image_path)
+        # Convert to PIL Image and then to 'RGBA'
+        tiff_image = Image.fromarray(np.uint8(tiff_image_array)).convert("RGBA")
+
+        # Load the mask image and convert it to grayscale
+        mask_image = Image.open(mask_image_path).convert("L")
+        mask_array = np.array(mask_image)
+        mask_array[mask_array > 0] = 255  # Ensure non-zero values are 255 (white)
+
+        # Find contours of the mask
+        contours = skimage.measure.find_contours(mask_array, level=128)
+
+        # Create an outline image with transparent background
+        outline_image = Image.new("RGBA", tiff_image.size, (0, 0, 0, 0))
+        draw = ImageDraw.Draw(outline_image)
+
+        for contour in contours:
+            # Swap x and y to match image coordinates
+            draw.line(
+                [(x, y) for y, x in np.round(contour).astype(int)],
+                fill=(0, 255, 0, 200),
+                width=2,
+            )
```

Comment:
Will this match a segmentation mask portion to the correct center x,y? Is x,y related to cells or nuclei?

Reply:
Thanks, good question - the `x, y` points here are parts of the segmentation mask outline (the green lines) which are calculated from the contours above. They don't necessarily relate to the center of any compartment objects and instead are part of the outline from the masks of the compartment object segmentations.